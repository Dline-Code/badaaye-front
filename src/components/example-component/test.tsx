/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render } from '../../utils/tests/wrapper'
import ExampleComponent from './'

describe('Example Component', () => {
  it('should render component correctly', () => {
    const { container, getByText } = render(<ExampleComponent type="button" />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText('My Button')).toBeInTheDocument()
  })

  it("should render component with  button type 'reset' ", () => {
    const { getByRole } = render(<ExampleComponent type="reset" />)

    expect(getByRole('button')).toHaveAttribute('type', 'reset')
  })

  it("should render component with  button type 'submit' ", () => {
    const { getByRole } = render(<ExampleComponent type="submit" />)

    expect(getByRole('button')).toHaveAttribute('type', 'submit')
  })

  it("should render component with  button type 'button' ", () => {
    const { getByRole } = render(<ExampleComponent type="button" />)

    expect(getByRole('button')).toHaveAttribute('type', 'button')
  })
})
