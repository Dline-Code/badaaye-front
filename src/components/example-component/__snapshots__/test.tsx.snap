// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component should render component correctly 1`] = `
<div
  class="styles__Wrapper-sc-1ltr66t-0 kUzvNb"
>
  <h1>
    My Button
  </h1>
  <button
    type="button"
  >
    Click here
  </button>
</div>
`;
